database/scrapers/event_calendar.ts

```diff
@@ -24,114 +25,137 @@ interface GoogleCalendarEventDto {
 }
 
 class CalendarParser {
-  private cursor: number;
+  private events: ICSValue[];
 
-  constructor(private calendarString: string) {
+  constructor(
+    private calendarString: string,
+    private task: TaskHandle,
+  ) {
     if (
       !this.calendarString.startsWith("BEGIN:VCALENDAR") ||
       !this.calendarString.trimEnd().endsWith("END:VCALENDAR")
     ) {
       throw new Error("Calendar invalid");
     }
-    this.cursor = calendarString.length - 16;
+    const calendarRoot = parseICS(calendarString, (message) =>
+      task.update(message),
+    ).VCALENDAR;
```

Comment:
`this.logger.warn` is probably the better warning logger here... but you'll need to pass it down from the scraper module